lib/todo/boards/board_lists/board_list_manager.ex

```diff
@@ -0,0 +1,85 @@
+defmodule Todo.Boards.BoardLists.BoardListManager do
+  alias Ecto.Multi
+  alias Todo.Boards.BoardList
+  alias Todo.Boards.BoardLists.BoardListPositions
+  alias Todo.Repo
+
+  @moduledoc """
+  This module handles creating and updating board lists. When a board list is created or updated,
+  the positions of the other board lists on that board will be re-ordered if required.
+  """
+
+  # @spec create(%{}, %Board{}) :: {:ok %BoardList{}} | {:error, String.t}
+  def create(%{"position" => position} = attrs, board) do
+    with %{valid?: true} = board_list_changeset <- insert_bl_changeset(attrs, board),
+         position <- String.to_integer(position),
+         {:ok, %{board_list: board_list}} <-
+           insert_board_list_and_positions(board_list_changeset, position, board.id) do
+      {:ok, board_list}
+    else
+      %{valid?: false} ->
+        {:error, "Invalid details"}
+
+      {:error, _} ->
+        {:error, "Unable to create board list"}
+    end
+  end
+
+  # @spec update(%{}, %BoardList{}) :: {:ok %BoardList{}} | {:error, String.t} | {:error, %Ecto.Changeset{}}
+  def update(
+        %{"position" => updated_position} = attrs,
+        %BoardList{board_id: board_id, position: current_position} = board_list
+      ) do
+    with %{valid?: true} = board_list_changeset <- update_bl_changeset(attrs, board_list),
+         updated_position <- String.to_integer(updated_position),
+         {:ok, %{board_list: board_list}} <-
+           update_board_list_and_positions(
+             board_list_changeset,
+             current_position,
+             updated_position,
+             board_id
+           ) do
+      {:ok, board_list}
+    else
```

Comment:
I think you need an extra `{error` case here. When you use Repo.transaction with a multi the error tuple is like this:

https://hexdocs.pm/ecto/Ecto.Repo.html#c:transaction/2

`{:error, Ecto.Multi.name(), any(), %{required(Ecto.Multi.name()) => any()}}`